addons/event/models/event_ticket.py

```diff
@@ -148,3 +148,8 @@ def is_launched(self):
             return ticket.start_sale_date <= current_date
         else:
             return True
+
+    @api.ondelete(at_uninstall=False)
+    def check_registrations_when_delete(self):
+        if self.registration_ids:
+            raise UserError(_("The operation cannot be completed: tickets having registrations cannot be deleted"))
```

Comment:
Could be great to give more hints to the user: like displaying all ticket names. Otherwise if someone unlinks 100 tickets but only 2 have registrations, it is not easy to find them back.

By the way, do we have a filter in ticket view allowing to filter out tickets with or without registrations ? Could be great to have it available if not.

Reply:
Added ticket names in the error message.

Tickets are only shown as part of an event and they don't have their own view, in this case, we can't have a filter no?
 
Furthermore, in the tickets treeview (which is inside the event view) the user has the possibility to sort tickets by the number of registrations, this gives him the possibility to easily see if a ticket has a registration or not.